### PR TITLE
Rename datasets

### PR DESCRIPTION
**Pull request by cikzh**
Renamed datasets,like `DefaultDS`, to `InternalDefaultDS`.
Their exposed counterparts are renamed from `ObservableDefaultDS` to `DefaultDS`.

This PR is best reviewed after #390 is merged

This rename wasn't much work, so if merging becomes hell, I'm happy to do the rename again in a different PR.

### CODE REVIEW

File: statime/Cargo.toml [hunk 1]
```diff
@@ -12,9 +12,11 @@ publish.workspace = true
 rust-version.workspace = true
 
 [features]
-default = ["std"]
+default = ["std", "serde", "tokio"]
 std = []
 fuzz = ["std"]
+serde = ["dep:serde"]
+tokio = ["dep:tokio"]
```
Comment (davidv1992) [inline statime/Cargo.toml:19]: What do we need the tokio dependence for in the library?

File: statime/src/datastructures/datasets/default.rs [hunk 2]
```diff
@@ -14,7 +14,8 @@ use crate::{
 /// those related to timebase, which is contained in the
 /// [TimePropertiesDS](crate::TimePropertiesDS) dataset.
 #[derive(Copy, Clone, Debug, Eq, PartialEq)]
-pub(crate) struct DefaultDS {
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub(crate) struct InternalDefaultDS {
```
Comment (davidv1992) [inline statime/src/datastructures/datasets/default.rs:18]: Why does this need serde (de)serialization?

File: statime/src/datastructures/messages/header.rs [hunk 3]
```diff
@@ -157,6 +157,7 @@ impl Default for Header {
 /// assert!(SdoId::try_from(0x1000).is_err());
 /// ```
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Default, Hash)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
```
Comment (davidv1992) [inline statime/src/datastructures/messages/header.rs:160]: SDO id should be twelve bits. This means the default serde serializer is OK, but the deserializer isn't and should be manually implemented.

File: statime/src/observability/mod.rs [hunk 4]
```diff
@@ -0,0 +1,119 @@
+//! Serializable implementations of datastructures to be used for observability
+#![allow(missing_docs)]
```
Comment (davidv1992) [inline statime/src/observability/mod.rs:2]: Think we should leave a TODO here that this file still needs better documentation.

File: statime/src/ptp_instance.rs [hunk 5]
```diff
@@ -148,19 +151,40 @@ impl PtpInstanceState {
 impl<F> PtpInstance<F> {
     /// Construct a new [`PtpInstance`] with the given config and time
     /// properties
-    pub fn new(config: InstanceConfig, time_properties_ds: TimePropertiesDS) -> Self {
-        let default_ds = DefaultDS::new(config);
+    pub fn new(config: InstanceConfig, time_properties_ds: InternalTimePropertiesDS) -> Self {
```
Comment (davidv1992) [inline statime/src/ptp_instance.rs:154]: Partially my fault, forgot that the we had time properties in the public interface here. Perhaps we should merge InternalTimePropertiesDS and TimePropertiesDS.

**Review by davidv1992 — CHANGES_REQUESTED** [review]
Few minor things that should still be cleaned up

File: statime/src/observability/mod.rs (hunk 4 above)
Comment (cikzh) [reply statime/src/observability/mod.rs:2]: I'll improve it right away

**Comment by cikzh** [comment]
This PR will be closed soon, since it'll be redone in #390

**Pull request closed without merging**